spikeinterface/sortingcomponents/waveforms/temporal_pca.py

```diff
@@ -0,0 +1,231 @@
+import pickle
+import json
+from pathlib import Path
+
+import numpy as np
+from sklearn.decomposition import IncrementalPCA
+
+from spikeinterface.sortingcomponents.peak_pipeline import PipelineNode, WaveformExtractorNode
+from spikeinterface.sortingcomponents.peak_detection import detect_peaks
+from spikeinterface.sortingcomponents.peak_selection import select_peaks
+from spikeinterface.postprocessing import compute_principal_components
+from spikeinterface.core import BaseRecording
+from spikeinterface.core.sparsity import ChannelSparsity
+from spikeinterface import extract_waveforms, NumpySorting
+from spikeinterface.core.job_tools import _shared_job_kwargs_doc
+from spikeinterface.core.recording_tools import get_channel_distances
+from .waveform_utils import to_temporal_representation, from_temporal_representation
+
+
+class TemporalPCBaseNode(PipelineNode):
+    def __init__(
+        self, recording: BaseRecording, parents: list[PipelineNode], model_folder_path: str, return_ouput=True
+    ):
+        """
+        Base class for PCA projection nodes. Contains the logic of the fit method that should be inherited by all the
+        child classess. The child should implement a compute method that does a specific operation
+        (e.g. project, denoise, etc)
+        """
+        PipelineNode.__init__(self, recording=recording, parents=parents, return_ouput=return_ouput)
+
+        self.model_folder_path = model_folder_path
+
+        if not Path(model_folder_path).is_dir() or model_folder_path is None:
+            exception_string = (
+                f"model_path folder is not a folder or does not exist. \n"
+                f"A model can be trained by using{self.__class__.__name__}.fit(...)"
+            )
+            raise TypeError(exception_string)
+
+        # Load the model and the time interval dict from the model_folder
+        model_path = Path(model_folder_path) / "pca_model.pkl"
+        with open(model_path, "rb") as f:
+            self.pca_model = pickle.load(f)
+        model_time_interval_path = Path(model_folder_path) / "time_interval.json"
```

Comment:
```suggestion
        model_time_interval_path = Path(model_folder_path) / "params.json"
```

I would call this params to unify across possible models